src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicSafeHandle.cs

```diff
@@ -25,7 +25,14 @@ internal unsafe class MsQuicSafeHandle : SafeHandle
 
     public override bool IsInvalid => handle == IntPtr.Zero;
 
-    public QUIC_HANDLE* QuicHandle => (QUIC_HANDLE*)DangerousGetHandle();
+    public QUIC_HANDLE* QuicHandle
+    {
+        get
+        {
+            ObjectDisposedException.ThrowIf(IsInvalid, this);
+            return (QUIC_HANDLE*)DangerousGetHandle();
+        }
+    }
```

Comment:
You don't need that property, [`SafeHandle.DangerousAddRef` throws if the handle is closed](https://github.com/dotnet/runtime/blob/941ec70ad37943762a18b2d1a4c45eb8e914cb4f/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeHandle.cs#L174-L177). Instead you can use `DangerousGetHandle()` according to the typical pattern of using `SafeHandles`.